src/Uno.UI/UI/Xaml/Media/Animation/Animators/FloatValueAnimator.iOSmacOS.cs

```diff
@@ -23,7 +23,12 @@ internal class FloatValueAnimator : IValueAnimator
 		private const int FrameRate = 50;
 
 		private long _numberOfFrames;
+#if __IOS__
 		private CADisplayLink _displayLink;
+#else
+		private NSTimer _timer;
+#endif
+
```

Comment:
Curious if you could use an approach as described here instead of an NSTimer: https://stackoverflow.com/a/14165842 ?

Reply:
I tried that, but I couldn't get the `CVDisplayLink` to run on the UI thread. It seems it is quite limited as compared to the `CADisplayLink` unfortunately

Reply:
Isn't that fairly common though to have a render loop runs on the render thread, not the UI thread? (similar to how UWP has a separate composition thread).
I can however see how that could cause an issue with sync, but you probably want animations be synced to render thread instead of the UI Thread?

Reply:
The common reason the timer is needed on the UI Thread here is for dependent animations mainly, to update affected dependency properties. Independent animations are generally handled differently (@dr1rrb will most probably have some feedback there).

Reply:
Aaah makes sense 👍

Reply:
The `GPUFloatAnimator` is there too, so it should work correctly 🙂

Reply:
Actually on iOS for independent animations we have 2 animators running at once:
1. The first which effectively animates the UI (the `GPUFloatAnimator`) 
1. The second which "fakely" animates the dependency property, and which obviously has to run on the UI thread (this `FloatValueAnimator`)

So it's indeed expected for the timer to tick on the UI thread here.

However I would argue about the need to always have those 2 animators instead of updating the DP only once at the end of the animation (or when paused). In my opinion we should add a feature flag for that, and disable it by default for perf considerations.

And about dependent animations, I think that they are not allowed on Uno yet, but they should indeed use this animator ... or actually a shared implementation which use the `DispatcherTimer`!

Note: On WASM animations are always running on UI thread, and on Android there is still no way to make them fully independent 😶

Reply:
Well, I just saw that we already have a [`DispatcherFloatAnimator`](https://github.com/unoplatform/uno/blob/master/src/Uno.UI/UI/Xaml/Media/Animation/Animators/DispatcherFloatAnimator.cs) @MartinZikmund did you tired to use it ? (Note: I didn't checked how complex it is to change the dependencies to use it instead of the `FloatValueAnimator`).